src/client/Client.cc

```diff
@@ -11403,6 +11403,12 @@ void Client::C_Write_Finisher::finish_io(int r)
 {
   bool fini;
 
+  // The objectcacher takes care of locking 'client_lock' before calling finish_io but not filer
+  // So lock only in case of filer write
+  std::unique_lock cl(clnt->client_lock, std::defer_lock);
+  if (!is_file_write)
```

Comment:
While this is fine, I want to propose that we use a separate callback that acquires `client_lock` for this case (filer write). The callback can wrap the call to `finish_io` with `client_lock` held.

Would that be a bit more nice way of implementing this?

Reply:
Yeah, that's cleaner. Done